### PR TITLE
csv readers take iterables as well as iterators

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by graingert** [comment]
See https://github.com/python/cpython/blob/3.6/Modules/_csv.c#L1564-L1569

File: Doc/library/csv.rst [hunk 1]
```diff
@@ -56,8 +56,9 @@ The :mod:`csv` module defines the following functions:
 .. function:: reader(csvfile, dialect='excel', **fmtparams)
 
    Return a reader object which will iterate over lines in the given *csvfile*.
-   *csvfile* can be any object which supports the :term:`iterator` protocol and returns a
-   string each time its :meth:`!__next__` method is called --- :term:`file objects
+   *csvfile* can be any object which supports the :term:`iterable` protocol and returns an
+   iterator that returns a string each time its :term`iterator`'s :meth:`!__next__` method
+   is called when its :meth:`!__iter__` method is called --- :term:`file objects
```
Comment (bitdancer) [inline Doc/library/csv.rst:61]: I would suggest instead changing the original with the following minimal change:
   `...can be any iterable object whose iterator returns a string each time its...`

**Review by bitdancer — CHANGES_REQUESTED** [review]

**Comment by bitdancer** [comment]
I don't think this is completely trivial, so you should open an issue for it.

File: Doc/library/csv.rst [hunk 2]
```diff
@@ -416,7 +417,7 @@ Reader objects have the following public attributes:
 
 .. attribute:: csvreader.line_num
 
-   The number of lines read from the source iterator. This is not the same as the
+   The number of lines read from the source iterable. This is not the same as the
```
Comment (vadmium) [inline Doc/library/csv.rst:420]: Why change this? If there is only one iterator involved, the original would be more precise.

Comment (bitdancer) [reply Doc/library/csv.rst:420]: That's a good point, iterator is more correct here.

**Comment by brettcannon** [comment]
To try and help move older pull requests forward, we are going through and backfilling 'awaiting' labels on pull requests that are lacking the label. Based on the current reviews, the best we can tell in an automated fashion is that a core developer requested changes to be made to this pull request.

If/when the requested changes have been made, please leave a comment that says, `I have made the requested changes; please review again`. That will trigger a bot to flag this pull request as ready for a follow-up review.

**Comment by graingert** [comment]
Sadly I am unable to honestly say that

I have made the requested changes; please review again.

**Comment by graingert** [comment]
However it was requested that I make an 'issue' first

**Comment by brettcannon** [comment]
Closing as the CLA has not been signed within the last month.

**Pull request closed without merging**